atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -185,8 +186,20 @@ public CompletableFuture<InstallResponse> onInstall(final InstallRequest request
                     .build()));
       }
 
-      pendingSnapshot =
-          raft.getPersistedSnapshotStore().newReceivedSnapshot(snapshotChunk.getSnapshotId());
+      try {
+        pendingSnapshot =
+            raft.getPersistedSnapshotStore().newReceivedSnapshot(snapshotChunk.getSnapshotId());
```

Comment:
🤔 Would it fail here or on the persist? If it fails here on start of the snapshot replication could we somehow disrupt the replication and say its ok we already have one?